app/src/main/java/a3/com/convo/adapters/FriendAdapter.java

```diff
@@ -53,61 +69,37 @@ public FriendAdapter(ArrayList<String> friends) {
     @Override
     public void onBindViewHolder(@NonNull final FriendAdapter.ViewHolder holder, int position) {
         holder.itemView.setBackgroundColor(selectedPos == position ? Color.rgb(229,229,229) : Color.TRANSPARENT);
-        // Get friends user ID's
-        final String friendId = myFriends.get(position);
-        final String profilePicPath = "/" + friendId + "/picture";
-
-        //Get friends profile picture
-//        GraphRequest picRequest = GraphRequest.newGraphPathRequest(AccessToken.getCurrentAccessToken(), "/" + friendId + "/picture", new GraphRequest.Callback() {
-//            @Override
-//            public void onCompleted(GraphResponse response) {
-//                try {
-//
-//                } catch (JSONException e) {
-//
-//                }
-//            }
-//        });
-
-        GraphRequest picRequest = GraphRequest.newMeRequest(
-                AccessToken.getCurrentAccessToken(),
-                new GraphRequest.GraphJSONObjectCallback() {
-                    @Override
-                    public void onCompleted(
-                            JSONObject json_object,
-                            GraphResponse response) {
-                        try {
-                            JSONObject picInfo = json_object.getJSONObject("picture").getJSONObject("data");
-                            String picURL = picInfo.optString("url");
-                            GlideApp.with(context)
-                                    .load(picURL)
-                                    .circleCrop()
-                                    .into(holder.ivFriend);
-                        } catch(Exception e){
-                            e.printStackTrace();
-                            Toast.makeText(context, "Pic URL not retrieved", Toast.LENGTH_SHORT).show();
-                        }
+        // Get friend TODO - Check if position in array is null/empty.
+        final String friend = myFriends.get(position);
+        // Find friend in background
+        ParseQuery<ParseUser> query = ParseUser.getQuery();
+        query.whereEqualTo(username, friend);
+        query.findInBackground(new FindCallback<ParseUser>() {
+            @Override
+            public void done(List<ParseUser> objects, ParseException e) {
+                // TODO - Should I check if empty or if null?
+                if (e == null && !objects.isEmpty()) {
```

Comment:
This should be `objects != null &&  !objects.isEmpty()`. This means we are checking if objects is not null or empty.